camera/src/test.py

```diff
@@ -0,0 +1,181 @@
+
+            if self.udpStreamEnabled:
+                pass
+
+    def stop(self):
+        self.running = False
+
+    def getFrame(self):
+        return self.frame
+
+    def openStream(self):
+        pass
+
+    def closeStream(self):
+        pass
+
+stream = WebcamStream()
+output_stream = FFMPEGOutputStream()
+
+# Load map between model output and labels
+with open(LabelMapPath, 'r') as map:
+    labels = [line.strip() for line in map.readlines()]
+
+print (labels)
+
+# Load model
+tflite_interpreter = tflite.Interpreter(CheckpointPath, num_threads=4)
+tflite_interpreter.allocate_tensors()
+
+input_details = tflite_interpreter.get_input_details()
+output_details = tflite_interpreter.get_output_details()
+
+input_data_idx = input_details[0]["index"]
+output_boxes_idx = output_details[1]["index"]
+output_classes_idx = output_details[3]["index"]
+output_scores_idx = output_details[0]["index"]
+
+input_mean = 127.5
+input_std = 127.5
+
+results = []
+
+while True:
+    frame = stream.getFrame().copy()
+    
+    # lower resolution
+    # resized_frame = cv2.resize(frame, (1138, 640), interpolation = cv2.INTER_AREA)
+    # crop to 640x640 size to input to model
+    # cropped_frame = resized_frame[0:640, 249:889]
+
+    start_time = time.perf_counter()
+    cropped_frame = cv2.resize(frame, (640, 640), interpolation = cv2.INTER_AREA)
+    rgb_frame = cv2.cvtColor(cropped_frame, cv2.COLOR_BGR2RGB)
+    # 1xHxWx3 (3 color channels)
+    input_data = np.expand_dims(rgb_frame, axis=0)
+    print(input_data.shape)
+    # Model is not quantazied, turn input values into floats
+    #input_data = (np.float32(input_data) - input_mean) / input_std
+    print("cv2: " + str(time.perf_counter() - start_time))
+    start_time = time.perf_counter()
+
+    tflite_interpreter.set_tensor(input_data_idx, input_data)
+    print("tflitea: " + str(time.perf_counter() - start_time))
+    tflite_interpreter.invoke()
+
+    print("tflite: " + str(time.perf_counter() - start_time))
+
+    results = []
+    output_boxes = tflite_interpreter.get_tensor(output_boxes_idx)[0]
+    output_classes = tflite_interpreter.get_tensor(output_classes_idx)[0]
+    output_scores = tflite_interpreter.get_tensor(output_scores_idx)[0]
+
+    for i in range(len(output_scores)):
+        box = output_boxes[i]
+        score = output_scores[i]
+        clasz = output_classes[i]
+
+        if score < 0.3: continue
+
+        box_y_min = int(max(1, box[0] * 1080))
+        box_x_min = int(max(1, box[1] * 1920))
+        box_y_max = int(min(1080,(box[2] * 1080)))
+        box_x_max = int(min(1920,(box[3] * 1920)))
+
+        lower_left = (box_x_min, box_y_min)
+        upper_right = (box_x_max, box_y_max)
+
+        results.append({
+            "box": (lower_left, upper_right),
+            "score": int(score * 100),
+            "name": labels[int(clasz)]
+        })
+
+    output_stream.submit(frame, results)
+
+    if cv2.waitKey(1) == ord('q'):
+        break
```

Comment:
When are we breaking out of the while loop? (in other words, help i don't understand what this if statement is saying)